### PR TITLE
Update Activity Shared Item Action Type

### PR DESCRIPTION
**Pull request by michaelwehrley**
Update Activity Shared Item Action Type

* Add callback that sets the the action type of an activity
  to be the status type of an activity.
* Update schema to have an activity model.
* Add spec to test model callback.

[#88917838]

### CODE REVIEW

**Comment by ericcf** [comment]
@michaelwehrley let's discuss. I really don't like this engine knowing about the Activity class.

**Comment by michaelwehrley** [comment]
A discussion sounds good to me! I feel the same way, but I am not sure of alternatives. Thanks!

**Comment by michaelwehrley** [comment]
@ericcf I made the dependency inject updates :vhs:

File: spec/models/social_networking/shared_item_spec.rb [hunk 1]
```diff
@@ -9,5 +9,20 @@ module SocialNetworking
       shared_item = SharedItem.create(item: goal, action_type: "action_type")
       expect(shared_item.participant_id).to eq(700_141_617)
     end
+
+    describe ".before_save" do
+      let(:activity) do
+        Activity.create!(participant_id: participants(:participant1).id)
+      end
+
+      describe ".set_action_type" do
```
Comment (ericcf) [inline spec/models/social_networking/shared_item_spec.rb]: @michaelwehrley instance methods typically are denoted with `#` as opposed to `.`.

**Comment by michaelwehrley** [comment]
How about private class methods?

**Comment by ericcf** [comment]
`set_action_type` is an instance, not class, method. It's usually considered bad practice to test private methods, so in this case rather than mentioning the method names, you might consider labelling based on the behavior? Something like

```
describe "when saving" do
...
end
```

**Comment by michaelwehrley** [comment]
@ericcf makes sense, thanks!

**Comment by ericcf** [comment]
:bread:

**Pull request merged**